`crucible-go` fails to build on `master`

Attempting to build `crucible-go` on a recent checkout of `master` yields this error:

```
[4 of 7] Compiling Lang.Crucible.Go.Overrides ( src/Lang/Crucible/Go/Overrides.hs, /home/rscott/Documents/Hacking/Haskell/sandbox/crucible/dist-newstyle/build/x86_64-linux/ghc-8.10.4/crucible-go-0.1.0.0/build/Lang/Crucible/Go/Overrides.o, /home/rscott/Documents/Hacking/Haskell/sandbox/crucible/dist-newstyle/build/x86_64-linux/ghc-8.10.4/crucible-go-0.1.0.0/build/Lang/Crucible/Go/Overrides.dyn_o )

src/Lang/Crucible/Go/Overrides.hs:120:33: error:
    Not in scope: type constructor or class ‘Crux.Model’
    Module ‘Crux.Types’ does not export ‘Model’.
    |
120 |              -> [(SomeOverride (Crux.Model sym) sym ext)]
    |                                 ^^^^^^^^^^
```

This is most likely due to #769. Fix incoming.

--- a/crucible-go/src/Lang/Crucible/Go/Overrides.hs
+++ b/crucible-go/src/Lang/Crucible/Go/Overrides.hs
@@ -117,7 +117,7 @@ do_assert = do
 
 go_overrides :: (IsSymInterface sym, 1 <= w)
              => NatRepr w
-             -> [(SomeOverride (Crux.Model sym) sym ext)]
+             -> [SomeOverride (p sym) sym ext]
 go_overrides w =
   [ mkSomeOverride "crucible" "FreshInt" Ctx.empty (BVRepr w) (fresh_int w)
   , mkSomeOverride "crucible" "FreshInt8" Ctx.empty

--- a/crucible-go/src/Lang/Crucible/Go/Simulate.hs
+++ b/crucible-go/src/Lang/Crucible/Go/Simulate.hs
@@ -142,15 +142,15 @@ mkGlobals sym globals =
             return $ insertGlobal glob zv state)
   emptyGlobals globals
 
-setupCrucibleGoCrux :: forall sym args.
+setupCrucibleGoCrux :: forall sym args p.
   (IsSymInterface sym, KnownRepr CtxRepr args)
   => Int                   -- ^ Machine word width
   -> Node P.SourcePos Main -- ^ Input program
   -> Int                   -- ^ Verbosity level
   -> sym                   -- ^ Simulator state
-  -> Crux.Model sym        -- ^ Personality
+  -> p sym                 -- ^ Personality
   -> C.RegMap sym args     -- ^ Arguments
-  -> IO (C.ExecState (Crux.Model sym) sym Go (C.RegEntry sym UnitType))
+  -> IO (C.ExecState (p sym) sym Go (C.RegEntry sym UnitType))
 setupCrucibleGoCrux machineWordWidth fwi verbosity sym p args = do
   setSimulatorVerbosity verbosity sym
 

--- a/crucible-go/tool/Main.hs
+++ b/crucible-go/tool/Main.hs
@@ -15,8 +15,6 @@ import Lang.Crucible.Simulator
 
 -- crux
 import qualified Crux
-import qualified Crux.Model   as Crux
-import qualified Crux.Types   as Crux
 
 -- Go
 import Language.Go.Parser
@@ -25,7 +23,7 @@ import Lang.Crucible.Go.Types
 import Paths_crucible_go (version)
 
 -- | A simulator context
-type SimCtxt sym = SimContext (Crux.Model sym) sym Go
+type SimCtxt sym = SimContext (Crux.Crux sym) sym Go
 
 data GoOptions = GoOptions { }
 
@@ -55,7 +53,7 @@ simulateGo copts _opts = Crux.SimulatorCallback $ \sym _maybeOnline -> do
    let regmap = RegMap Ctx.Empty
 
    -- Set up initial crucible execution state
-   initSt <- setupCrucibleGoCrux 32 fwi verbosity sym Crux.emptyModel regmap
+   initSt <- setupCrucibleGoCrux 32 fwi verbosity sym Crux.CruxPersonality regmap
 
    -- TODO: add failure explanations
    let explainFailure _evalFn _gl = return mempty

